preload.php

```diff
@@ -2,240 +2,263 @@
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/ErrorHandler/Collecting.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/ParserAbstract.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeVisitorAbstract.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Comment/Doc.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/TraitUse.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/Function_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/Trait_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/Property.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/Namespace_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/Use_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/Class_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/Interface_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/Declaration.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/Param.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/TraitUseAdaptation.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/FunctionLike.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Builder/Method.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Comment.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeTraverserInterface.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/ConstExprEvaluationException.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeVisitor.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/BuilderFactory.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/BuilderHelpers.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Error.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/ParserFactory.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeAbstract.php';
+final class RectorVendor
+{
+    /**
+     * @var string
+     */
+    private static $rootPath;
+
+    public static function getRootPath()
+    {
+        if (self::$rootPath) {
+            return self::$rootPath;
+        }
+
+        if (file_exists(__DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Lexer.php')) {
+            self::$rootPath = __DIR__;
+            return self::$rootPath;
+        }
+
+        self::$rootPath = __DIR__ . '/../../..';
+        return self::$rootPath;
+    }
+}
+
+RectorVendor::getRootPath() . '/vendor/nikic/php-parser/lib/PhpParser/Lexer.php';
```

Comment:
Did you skip `require_once` on purpouse?

Reply:
Ooops, I missed that. As @TomasVotruba's comment https://github.com/rectorphp/rector/pull/6324#issuecomment-831855738 . It need only loaded on tests / prefixed version.